packages/reakit/src/Dialog/__utils/useEventListenerOutside.ts

```diff
@@ -58,6 +58,9 @@ export function useEventListenerOutside(
         return;
       }
 
+      // Fix for https://github.com/reakit/reakit/issues/619
+      if (document.hasFocus()) return;
```

Comment:
If it returns, it won't close the dialog. So this is basically saying that if anything in the document has focus, it shouldn't close the dialog.

Also, `document` should use the `document` constant that is defined below this handler function (just moving it before the function should be enough).